Mathlib/Topology/Separation/DisjointCover.lean

```diff
@@ -58,3 +59,110 @@ lemma exists_finite_nonempty_disjoint_clopen_cover (hU : IsOpenCover U) :
   · exact hWd.comp_of_injective <| Subtype.val_injective.comp t.equivFin.symm.injective
 
 end TopologicalSpace.IsOpenCover
+
+namespace TopologicalSpace
+variable {X : Type*} [TopologicalSpace X] {S : Set (X × X)}
+
+/-- If `S` is any neighbourhood of the diagonal in a topological space `X`, any point of `X` has an
+open neighbourhood `U` such that `U ×ˢ U ⊆ S`. -/
+lemma exists_open_prod_subset_of_mem_nhds_diagonal (hS : S ∈ nhdsSet (diagonal X)) (x : X) :
+    ∃ U : Set X, IsOpen U ∧ x ∈ U ∧ U ×ˢ U ⊆ S := by
+  have : S ∈ 𝓝 (x, x) := mem_nhdsSet_iff_forall.mp hS _ rfl
+  obtain ⟨u, v, huo, hux, hvo, hvx, H⟩ := by rwa [mem_nhds_prod_iff'] at this
+  exact ⟨_, huo.inter hvo, ⟨hux, hvx⟩, fun p hp ↦ H ⟨hp.1.1, hp.2.2⟩⟩
+
+variable [CompactSpace X]
+
+/-- If `S` is any neighbourhood of the diagonal in a compact topological space `X`, then there
+exists a finite cover of `X` by opens `U i` with `U i ×ˢ U i ⊆ S` for all `i`.
+
+That the indexing set is a finset of `X` is an artifact of the proof; it could be any finite type.
+-/
+lemma exists_finite_open_cover_prod_subset_of_mem_nhds_diagonal_of_compact
+    (hS : S ∈ nhdsSet (.diagonal X)) :
+    ∃ (t : Finset X) (U : t → Opens X), IsOpenCover U ∧ ∀ i, (U i : Set X) ×ˢ U i ⊆ S := by
```

Comment:
```suggestion
    ∃ (D : Finset X) (U : D → Opens X), IsOpenCover U ∧ ∀ i, (U i : Set X) ×ˢ U i ⊆ S := by
```
I am suggesting this change simply to be more uniform with the two lemmas below.